src/bucket/BucketList.cpp

```diff
@@ -657,6 +658,29 @@ BucketList::levelShouldSpill(uint32_t ledger, uint32_t level)
             ledger == roundDown(ledger, levelSize(level)));
 }
 
+// Update period for curr buckets is the incoming spill frequency of the level
+// which the bucket is in. For snap buckets, the update period is the incoming
+// spill frequency of the level below.
+// incoming_spill_frequency(i) = 2^(2i - 1) for i > 0
+// incoming_spill_frequency(0) = 1
+uint32_t
+BucketList::bucketChangeRate(uint32_t level, bool isCurr)
```

Comment:
As much as I believe this is _likely_ correct, literally ever time we've done BL arithmetic in the past we've got it slightly wrong; could you add some explicit test vectors on this calculation? just like .. either fish out some specific spills that happened in the history and confirm that they happened this-number-of-ledgers apart, or write a test that just calculates these numbers and then drives the BL forward and checks that the spills happen at the frequency predicted?